shared/query.go

```diff
@@ -31,6 +35,46 @@ var UploadsURL = url.URL{
 	Host:   "uploads.mangadex.org",
 }
 
+
+// cache holds cached responses for image and API requests
+var cache, _ = bigcache.New(context.Background(), bigcache.Config{
```

Comment:
Not certain if this cache system is entirely necessary, but I'm alright with including it and I like this implementation. BigCache seems like a nice library and a good choice for this.

I feel like it should be the client reader's job to do caching, not this proxy server.
But we're generating several-megabyte files so it might be a good idea to cache here anyway. Also it's nicer to MangaDex.